tests/unit/session/test_session.py

```diff
@@ -63,15 +65,36 @@ def test_read_gbq_not_found_tables(not_found_table_id):
     ],
 )
 def test_read_gbq_external_table_no_drive_access(api_name, query_or_table):
-    bqclient = mock.create_autospec(google.cloud.bigquery.Client, instance=True)
-    bqclient.project = "test-project"
-    bqclient.get_table.side_effect = google.api_core.exceptions.Forbidden(
-        "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials."
-    )
-    session = resources.create_bigquery_session(bqclient=bqclient)
+    session = resources.create_bigquery_session()
+    mock_query_job = session.bqclient.query.return_value
+
+    def query_mock(query, *args, **kwargs):
+        query_job = copy.copy(mock_query_job)
+
+        if query.lstrip().startswith("SELECT *"):
+            raise google.api_core.exceptions.Forbidden(
+                "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials."
+            )
+
+        if query.startswith("SELECT CURRENT_TIMESTAMP()"):
```

Comment:
Nit: This logic doesn't really relate to what you're trying to test. Maybe there's a way we can move this to the `create_bigquery_session` helper?

Reply:
Thanks, updated.